internal/server/http/server.go

```diff
@@ -29,21 +29,28 @@ type ServerOpt func(s *Server)
 
 // Server definition of mock server
 type Server struct {
-	router     *mux.Router
-	httpServer *http.Server
-	proxy      *Proxy
-	secure     bool
-	imposterFs ImposterFs
+	router           *mux.Router
+	httpServer       *http.Server
+	proxy            *Proxy
+	secure           bool
+	imposterFs       ImposterFs
+	corsOptions      []handlers.CORSOption
+	logLevel         int
+	dumpRequestsPath string
+	dumpCh           chan *RequestData
 }
 
 // NewServer initialize the mock server
-func NewServer(r *mux.Router, httpServer *http.Server, proxyServer *Proxy, secure bool, fs ImposterFs) Server {
+func NewServer(r *mux.Router, httpServer *http.Server, proxyServer *Proxy, secure bool, fs ImposterFs, CORSOptions []handlers.CORSOption, logLevel int, dumpRequestsPath string) Server {
```

Comment:
If we're going to keep adding more and more options (three this time), I'd love to see options as functional arguments, in a separate package. Especially for those that aren't required, the rest can be kept as regular args.

Reply:
Added a separate serverconfig package to handle extra args.